docs/upgrade_guide.md

```diff
@@ -0,0 +1,30 @@
+# Upgrade Guide
+
+This guide describes how to upgrade Open Service Mesh (OSM).
+
```

Comment:
We should add somewhere at the top that control plane upgrades are disruptive. It may make sense for an end user to mirror their environment on a different cluster, re-direct traffic to the mirrored environment before considering a control plane upgrade (especially one that involves CRDs)

Reply:
Moved that section up towards the top